harvard-huit/Harvard-Geospatial-Library
# About this item

**Fixing JS bug that attempted to create duplicate leaflet maps.**
* * *

**JIRA Ticket**: none

* Relevant discussion on Stackoverflow with solution: https://github.com/Leaflet/Leaflet/issues/3962#issuecomment-384510881

# What does this Pull Request do?
When viewing detail pages when the JavaScript is compressed (as it would be in QA and Production environments), there is a giant JavaScript error in the console ("Uncaught Error: Map container is already initialized.") and scripts below that error do not fire (For example, the titles under the "About this item" heading in the right rail do not display in the browser).

# How should this be tested?

* To reproduce the problem, temporarily switch your local environment to compress JS and restart the Docker container (you do not need to fully rebuild).  To do this change the value in this line to `false`: https://github.com/harvard-huit/Harvard-Geospatial-Library/blob/main/config/environments/development.rb#L54 
* After changing that setting, You can see the error in the console and the titles not load on a detail page.  Additionally if you click on a map tile the index explorer will open but you will not be able to close it. This detail page is a good example: https://localhost:3001/catalog/harvard-g5831-r3-s20-f7-index-gr
* After incorporating this PR / Branch you can restart your container (you do not need to fully rebuild) then refresh the page. The JS error in the console will disappear and the rest of the JS actions on the page will work properly.

# Test coverage
Yes/No: Are changes in this pull-request covered by:
- unit tests? No
- integration tests? No

# Interested parties
@mferrarini @enriquediaz
